Megaparsec/Errors.lean

```diff
@@ -58,6 +69,36 @@ inductive ErrorFancy (E : Type u) where
   | indent (ord : Ordering) (fromPos : Pos) (uptoPos : Pos)
   | custom (e : E)
 
+private def compareOrderings : Ordering → Ordering → Ordering
+  | .gt, .gt => .eq
+  | .gt,   _ => .gt
+  | .eq, .gt => .lt
+  | .eq, .eq => .eq
+  | .eq, .lt => .gt
+  | .lt, .lt => .eq
+  | .lt,   _ => .lt
+
+private def compareIndents : (Ordering × Pos × Pos) → (Ordering × Pos × Pos)
+                           → Ordering
+  | (o1, f1, u1), (o2, f2, u2) =>
+    match compare u1 u2, compare f1 f2 with
+                  -- we compare orderings as that's the last differing parameter
```

Comment:
I still don't understand it. Can you remind me the design of indent triples (as in, what does the first element in the triple mean?).

I understand the logic but don't understand the semantics.

Reply:
It's mostly for formatting AFAICT, or rather for better location pointing. The indent triple is `fromPos`, `uptoPos`, and this ordering, and it's used in the `ToString` instance to say whether the correct indentation should be greater than `fromPos` or equal or less. So it's IMO the least important parameter.

Reply:
I still don't understand it, sorry. I think it warrants a docstring, explaining what happens if the indentation is bigger, what if less than, what if equal.